src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php

```diff
@@ -196,48 +168,21 @@ public function resolveValue($value, array $resolving = array())
      * @throws ParameterNotFoundException          if a placeholder references a parameter that does not exist
      * @throws ParameterCircularReferenceException if a circular reference if detected
      * @throws RuntimeException                    when a given parameter has a type problem.
+     *
+     * @deprecated since 3.1, to be removed in 4.0. Use {@link static::resolveValue()} instead.
      */
     public function resolveString($value, array $resolving = array())
     {
-        // we do this to deal with non string values (Boolean, integer, ...)
-        // as the preg_replace_callback throw an exception when trying
-        // a non-string in a parameter value
-        if (preg_match('/^%([^%\s]+)%$/', $value, $match)) {
-            $key = strtolower($match[1]);
-
-            if (isset($resolving[$key])) {
-                throw new ParameterCircularReferenceException(array_keys($resolving));
-            }
-
-            $resolving[$key] = true;
-
-            return $this->resolved ? $this->get($key) : $this->resolveValue($this->get($key), $resolving);
-        }
-
-        return preg_replace_callback('/%%|%([^%\s]+)%/', function ($match) use ($resolving, $value) {
-            // skip %%
-            if (!isset($match[1])) {
-                return '%%';
-            }
-
-            $key = strtolower($match[1]);
-            if (isset($resolving[$key])) {
-                throw new ParameterCircularReferenceException(array_keys($resolving));
-            }
-
-            $resolved = $this->get($key);
-
-            if (!is_string($resolved) && !is_numeric($resolved)) {
-                throw new RuntimeException(sprintf('A string value must be composed of strings and/or numbers, but found parameter "%s" of type %s inside string value "%s".', $key, gettype($resolved), $value));
-            }
+        @trigger_error(sprintf('%s is deprecated since version 3.1 and will be removed in 4.0. Use %s::resolveValue() instead.', __METHOD__, get_class($this)), E_USER_DEPRECATED);
```

Comment:
why using `__METHOD__`, then `get_class($this)`?
what about : `The %s method is deprecated since version 3.1 and will be removed in 4.0. Use the resolveValue() method instead.`